SleepEngine/Engine/Jobs/JobThread.cpp

```diff
@@ -0,0 +1,45 @@
+#include "stdafx.h"
+#include "JobThread.h"
+
+BEGIN_NAMESPACE_SLEEP
+
+JobThread::JobThread(JobQueue& jobQueue, Event& workAvailable, std::atomic<bool> const& shutdownRequested)

```

Comment:
why passing atomic bool by reference?
Can't you pass a simple bool?

Reply:
It's modified in the main thread at the end, and read in parallel (possibly at the same time) by all other threads, so there can be race condition